llext: build error in `gen_device_deps.py` when exporting devices on `intel_adsp/ace15_mtpm`

**Describe the bug**
When building the `llext.simple.loader_build` test (under `tests/subsys/llext/simple`) with board `intel_adsp/ace15_mtpm`, the `gen_device_deps.py` errors out when unpacking data from the pre0 ELF file. It seems like the exported device symbols for LLEXT, e.g. `__device_dts_ord_*_sym`, are causing the failure.

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b intel_adsp/ace15_mtpm tests/subsys/llext/simple/
2. See error

**Expected behavior**
No error.

**Impact**
Annoyance for now. But a full CI run with the board is causing issue.

**Logs and console output**
```
[222/233] Generating device_deps.c
FAILED: zephyr/device_deps.c /tmp/build/zephyr/device_deps.c 
cd /tmp/build/zephyr && zephyr/venv/zephyr/bin/python3 zephyr/rtos/zephyr/scripts/build/gen_device_deps.py --output-source device_deps.c --output-graphviz dev_graph.dot --num-dynamic-devices 0 --kernel /tmp/build/zephyr/zephyr_pre0.elf --zephyr-base zephyr/rtos/zephyr --start-symbol _device_list_start
Traceback (most recent call last):
  File "zephyr/rtos/zephyr/scripts/build/gen_device_deps.py", line 168, in <module>
    main()
  File "zephyr/rtos/zephyr/scripts/build/gen_device_deps.py", line 128, in main
    parsed_elf = ZephyrElf(args.kernel, edt, args.start_symbol)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 130, in __init__
    self._device_parse_and_link()
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 242, in _device_parse_and_link
    self._object_find_named('__device_', _on_device)
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 186, in _object_find_named
    cb(sym)
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 241, in _on_device
    self.devices.append(Device(self, sym))
                        ^^^^^^^^^^^^^^^^^
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 109, in __init__
    self.obj_ordinals = self._data_native_read(ordinal_offset)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "zephyr/rtos/zephyr/scripts/build/elf_parser.py", line 32, in _data_native_read
    return struct.unpack(format, self.data[offset:offset + size])[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
struct.error: unpack requires a buffer of 4 bytes
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA: 5e2203a6cdd0daf395bb3825d416843ecdc40629

**Additional context**
If changing line 185 in `scripts/build/elf_parser.py` from:
```
if sym.name.startswith(prefix):
```
to
```
if sym.name.startswith(prefix) and not sym.name.endswith('_sym'):
```
Then the build was fine. Not sure this would be the correct fix though.

--- a/include/zephyr/llext/symbol.h
+++ b/include/zephyr/llext/symbol.h
@@ -93,7 +93,7 @@ struct llext_symtable {
 #define Z_LL_EXTENSION_SYMBOL(x)						\
 	static const struct llext_const_symbol					\
 			Z_GENERIC_SECTION(".exported_sym") __used		\
-			x ## _sym = {						\
+			__llext_sym_ ## x = {					\
 		.name = STRINGIFY(x), .addr = (const void *)&x,			\
 	}
 #else
@@ -124,13 +124,15 @@ struct llext_symtable {
 #define Z_EXPORT_SYMBOL(x)							\
 	static const char Z_GENERIC_SECTION("llext_exports_strtab") __used	\
 		x ## _sym_name[] = STRINGIFY(x);				\
-	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, x ## _sym) = {	\
+	static const STRUCT_SECTION_ITERABLE(llext_const_symbol,                \
+					     __llext_sym_ ## x) = {	        \
 		.name = x ## _sym_name, .addr = (const void *)&x,		\
 	}
 #elif defined(CONFIG_LLEXT)
 /* LLEXT application: export symbols */
 #define Z_EXPORT_SYMBOL(x)							\
-	static const STRUCT_SECTION_ITERABLE(llext_const_symbol, x ## _sym) = {	\
+	static const STRUCT_SECTION_ITERABLE(llext_const_symbol,                \
+					     __llext_sym_ ## x) = {	        \
 		.name = STRINGIFY(x), .addr = (const void *)&x,			\
 	}
 #else

